pkg/client/cache/shared_informer.go

```diff
@@ -273,6 +276,7 @@ func (s *sharedIndexInformer) HandleDeltas(obj interface{}) error {
 	for _, d := range obj.(Deltas) {
 		switch d.Type {
 		case Sync, Added, Updated:
+			s.cacheMutationDetector.AddObject(d.Object)
```

Comment:
> Don't we also want to remove on Deleted?

Deletion doesn't guarantee references are gone.  If we end up with memory pressure problems during runs, we could consider that.